docs/modules/engines/pages/introduction.adoc

```diff
@@ -0,0 +1,21 @@
+
+= Correlation Engines
+
+Correlation engines are the underlying processes that drive ALEC.
+Two engines are implemented by default: xref:clustering.adoc[Clustering] and xref:deeplearning.adoc[Deep Learning].
+Both extend the cluster engine, which builds network topology graphs based on information provided by your datasource (see xref:datasources:overview.adoc[ALEC Datasources]).
+
+== Cluster engine
+
+The cluster correlation engine builds a network topology graph.
+The graph is based on network inventory information provided by the datasource, and is persisted in local memory.
+Inventory objects are added to the graph as vertices, and relationships among them are added as edges.
+Alarms are then attached to the vertex of the inventory object that they reference.
+
+On every refresh interval (which defaults to 30 seconds), ALEC runs a clustering algorithm against all alarms on the graph.
```

Comment:
I assume this interval is set in the OpenNMS core, so probably no need to document, but let's verify.